### PR TITLE
[tests-only][full-ci] Extend space download tests coverage

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR is follow-up PR of https://github.com/owncloud/ocis/pull/6683 to add more tests related to space download 

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/6606

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiDownloads/spaceDownload.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      | file1.txt        | some data         |
+      | .space/readme.md | space description |
+    Examples:
+      | role    |
+      | manager |
+      | editor  |
+      | viewer  |
+
+
+  Scenario Outline: admin and space admin tries to downloads other user space
```
Comment (phil-davis) [inline tests/acceptance/features/apiDownloads/spaceDownload.feature:49]:
```suggestion
  Scenario Outline: admin and space admin try to download a space that they do not have access to
```

Maybe this is a better description of the scenario?

**Review by phil-davis — COMMENTED** [review]
Code and tests LGTM. Just one suggestion for a scenario description.

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -3242,18 +3242,20 @@ public function theUserShouldHaveSpaceWithRecipient(
...
 	 *
 	 * @return void
 	 *
 	 * @throws GuzzleException
 	 */
-	public function userDownloadsTheSpaceUsingTheWebdavApi(string $user, string $spaceName):void {
-		$space = $this->getSpaceByName($user, $spaceName);
-		$url = $this->featureContext->getBaseUrl() . '/archiver?id=' . $space["id"];
+	public function userDownloadsTheSpaceUsingTheWebdavApi(string $user, string $spaceName, string $owner = ''):void {
+		$space = $this->getSpaceByName(($owner !== '') ? $owner : $user, $spaceName);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3257]:
```suggestion
		$space = $this->getSpaceByName($owner ?? $user, $spaceName);
```

File: tests/acceptance/features/apiDownloads/spaceDownload.feature [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      | file1.txt        | some data         |
+      | .space/readme.md | space description |
+    Examples:
+      | role    |
+      | manager |
+      | editor  |
+      | viewer  |
+
+
+  Scenario Outline: admin and space admin try to download a space that they do not have access to
```
Comment (saw-jan) [inline tests/acceptance/features/apiDownloads/spaceDownload.feature:49]: scenario tests one user at a time
```suggestion
  Scenario Outline: admin/space-admin tries to download a space that they do not have access to
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 4]
```diff
@@ -3242,18 +3242,20 @@ public function theUserShouldHaveSpaceWithRecipient(
...
 	 *
 	 * @return void
 	 *
 	 * @throws GuzzleException
 	 */
-	public function userDownloadsTheSpaceUsingTheWebdavApi(string $user, string $spaceName):void {
-		$space = $this->getSpaceByName($user, $spaceName);
-		$url = $this->featureContext->getBaseUrl() . '/archiver?id=' . $space["id"];
+	public function userDownloadsTheSpaceUsingTheWebdavApi(string $user, string $spaceName, string $owner = ''):void {
+		$space = $this->getSpaceByName($owner ?? $user, $spaceName);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3257]: use ternary operator. you have empty string as default for `$owner` which will evaluate to true with `??` (nullish coalescing )
```suggestion
		$space = $this->getSpaceByName($owner ?: $user, $spaceName);
```

**Review by saw-jan — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiDownloads/spaceDownload.feature [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      | file1.txt        | some data         |
+      | .space/readme.md | space description |
+    Examples:
+      | role    |
+      | manager |
+      | editor  |
+      | viewer  |
+
+
+  Scenario Outline: admin/space-admin triesO to download a space that they do not have access to
```
Comment (phil-davis) [inline tests/acceptance/features/apiDownloads/spaceDownload.feature:49]:
```suggestion
  Scenario Outline: admin/space-admin tries to download a space that they do not have access to
```

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@amrita-shrestha this PR is to `stable-3.0`
Will you forward-port it to master?

**Review by saw-jan — APPROVED** [review]

**Pull request merged**